src/kinetics/ReactionData.cpp

```diff
@@ -267,4 +262,74 @@ void ChebyshevData::restore()
     m_pressure_buf = -1.;
 }
 
+CoverageData::CoverageData()
+    : sqrtT(NAN)
+{
+}
+
+void CoverageData::update(double T)
+{
+    throw CanteraError("CoverageData::update",
+        "Missing state information: 'CoverageData' requires species coverages.");
+}
+
+void CoverageData::update(double T, const vector_fp& values)
+{
+    warn_user("CoverageData::update",
+        "This method does not update the site density.");
+    ReactionData::update(T);
+    sqrtT = sqrt(T);
+    if (coverages.size() == 0) {
+        coverages = values;
+        logCoverages.resize(values.size());
+    } else if (values.size() == coverages.size()) {
+        std::copy(values.begin(), values.end(), coverages.begin());
+    } else {
+        throw CanteraError("CoverageData::update",
+            "Incompatible lengths of coverage arrays: received {} elements while "
+            "{} are required.", values.size(), coverages.size());
+    }
+    for (size_t n = 0; n < coverages.size(); n++) {
+        logCoverages[n] = std::log(std::max(coverages[n], Tiny));
+    }
+}
+
+bool CoverageData::update(const ThermoPhase& phase, const Kinetics& kin)
+{
+    int mf = phase.stateMFNumber();
+    double T = phase.temperature();
+    bool changed = false;
+    const auto& surf = dynamic_cast<const SurfPhase&>(
+        kin.thermo(kin.surfacePhaseIndex()));
+    double site_density = surf.siteDensity();
+    if (density != site_density) {
+        density = surf.siteDensity();
+        changed = true;
+    }
+    if (T != temperature) {
+        ReactionData::update(T);
+        sqrtT = sqrt(T);
+        changed = true;
+    }
+    if (changed || mf != m_state_mf_number) {
+        phase.getActivityConcentrations(coverages.data());
+        surf.getCoverages(coverages.data());
+        for (size_t n = 0; n < coverages.size(); n++) {
+            logCoverages[n] = std::log(std::max(coverages[n], Tiny));
+        }
+        for (size_t n = 0; n < kin.nPhases(); n++) {
+            kin.thermo(n).getPartialMolarEnthalpies(
+                grt.data() + kin.kineticsSpeciesIndex(0, n));
```

Comment:
Sorry I hadn't noticed this before, but the variable `BlowersMaselData::grt` should probably be named something like `partial_molar_enthalpies` if that's what it holds. The name `grt` seems like it should only be a shorthand for nondimensional Gibbs free energy.